govcd/lbserverpool.go

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// CreateLBServerPool creates a load balancer server pool based on mandatory fields. It is a synchronous
+// operation. It returns created object with all fields (including ID) populated or an error.
+// Name and Algorithm fields must be populated.
+func (eGW *EdgeGateway) CreateLBServerPool(lbPoolConfig *types.LBPool) (*types.LBPool, error) {
+	if err := validateCreateLBServerPool(lbPoolConfig); err != nil {
+		return nil, err
+	}
+
+	httpPath, err := eGW.buildProxiedEdgeEndpointURL(types.LBServerPoolPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+	// We expect to get http.StatusCreated or if not an error of type types.NSXError
+	resp, err := eGW.client.ExecuteRequestWithCustomError(httpPath, http.MethodPost, types.AnyXMLMime,
+		"error creating load balancer server pool: %s", lbPoolConfig, &types.NSXError{})
+	if err != nil {
+		return nil, err
+	}
+	location := resp.Header.Get("Location")
+
+	// Last element in location header is the server pool ID
+	// i.e. Location: [/network/edges/edge-3/loadbalancer/config/pools/pool-7]
+	if location == "" {
+		return nil, fmt.Errorf("unable to retrieve ID for new load balancer server pool with name %s", lbPoolConfig.Name)
+	}
+	splitLocation := strings.Split(location, "/")
+	lbPoolID := splitLocation[len(splitLocation)-1]
+
+	readPool, err := eGW.ReadLBServerPool(&types.LBPool{ID: lbPoolID})
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve lb server pool with ID (%s) after creation: %s", lbPoolID, err)
+	}
+	return readPool, nil
+}
+
+// ReadLBServerPool is able to find the types.LBPool type by Name and/or ID.
+// If both - Name and ID are specified it performs a lookup by ID and returns an error if the specified name and found
+// name do not match.
+func (eGW *EdgeGateway) ReadLBServerPool(lbPoolConfig *types.LBPool) (*types.LBPool, error) {
```

Comment:
Why we don't use `id, name string` ? this seems overcomplicated

Reply:
I've created these as flexible functions even for future. Having them it is very cheap and easy to create wrappers with less inputs.

Reply:
well, currently it's ugly to use: `readLBPool, err := edgeGateway.ReadLBServerPool(&types.LBPool{ID: d.Id()})`

Reply:
I can add wrappers if we agree to do so.

Reply:
Added for both - Server Pool and Service Monitor.